librz/io/p/io_srec.c

```diff
@@ -0,0 +1,810 @@
+// SPDX-FileCopyrightText: 2013-2017 pancake <pancake@nopcode.org>
+// SPDX-FileCopyrightText: 2013-2017 fenugrec <fenugrec@users.sourceforge.net>
+// SPDX-License-Identifier: LGPL-3.0-only
+// Contributed by manouser <manolache.alexandru8@gmail.com>
+
+/*
+*** SREC format description : every line follows this pattern
+	S / Type / Bytecount / Address / Data / Checksum
+
+	There are a bunch of types, from which one (S4) is reserved
+
+//sauce : https://en.wikipedia.org/wiki/SREC_(file_format)
+
+**** example records, not sure about new adresses tho
+	every other function besides SREC parse is from io_ihex.c
+
+S00F000068656C6C6F202020202000003C
+S11F00007C0802A6900100049421FFF07C6C1B787C8C23783C6000003863000026
+S11F001C4BFFFFE5398000007D83637880010014382100107C0803A64E800020E9
+S5030003F9
+S9030000FC
+*/
+
+#include "rz_io.h"
+#include "rz_lib.h"
+#include "rz_util.h"
+#include <limits.h> //for INT_MAX
```

Comment:
you don't need int_max, please use the values defined here: https://github.com/rizinorg/rizin/blob/a0be4e2e645d55c718cfb14f33c0163d25960260/librz/include/rz_types_base.h#L85 (STxx stands for SIGNED TYPE, meanwhile UTxx stands for UNSIGNED TYPE)

Reply:
it's already included with `rz_util.h`

Reply:
I will come back with the modifications tonight. Thank you for the review!